src/backprojection.cu

```diff
@@ -7,6 +7,30 @@
 #include "texture.h"
 #include "backprojection.h"
 
+namespace
+{
+template<typename T>
+__device__ T toType(float);
+
+template<>
+__device__ float toType(float f)
```

Comment:
The cuda compiler on windows doesn't allow implicit casts from 32bit floats to 16bit floats. In `radon_backward_kernel`, for example, `accumulator` is a 32bit float array, but `output` might be 16bit float. `toType` casts 32bit floats to the target type.

Reply:
I just found out that `CUDAExtension` defines `__CUDA_NO_HALF_CONVERSIONS__` for nvcc, which deactivates these implicit conversions. Since you don't compile the cuda files with `CUDAExtension`, you didn't get these errors.
There are 2 possibilities now:
1. keep the templated casts in the cuda files
2. undefine `__CUDA_NO_HALF_CONVERSIONS__` for `CUDAExtension`

Reply:
I am ok with keeping the templated casts. Please put them in a separate file in order to avoid duplication of code